pkg/lib/cockpit.d.ts

```diff
@@ -368,4 +368,18 @@ declare module 'cockpit' {
 
     export let localStorage: Storage;
     export let sessionStorage: Storage;
+
+    export interface HttpOptions {
+        address: string;
+        port: number;
+        headers: { [key: string]: string };
+      }
+    
+      export interface HttpInstance {
+        request(options: { path: string; method: string; body: string }): Promise<string>;
```

Comment:
All options are, well, optional, and it's missing "params" and "headers". See https://cockpit-project.org/guide/latest/cockpit-http.html#cockpit-http-request .

The return value is more tricky as well: If the constructor has the `binary: true` option, the promise delivers an UInt8Array.

There is precedent in this file, e.g. for `function channel()` or `spawn()` how to do this.